RationaleMCP/0035/Readme.md

```diff
@@ -0,0 +1,129 @@
+* Where the translation files to be stored
+* How does a GNU gettext translation template for a library look like and which kind of descriptive texts shall be included?
+* Which strings should be translated:
+
+The precise updated text of the specification is part of this branch/pull-request: Yes.
+
+# Backwards Compatibility
+The proposal is backwards compatible. All multilingual text changes can be applied without effect at existing Modelica code. 
+After the introduction of translation files for libraries, this files need to be adjusted with every library change which adds new commented parts or changes existing comments.
+
+# Tool Implementation
+
+## Experience with Prototype
+### Implementation steps
+#### 1. Read out a translation template file
+Create at first a translation template file \*.pot for a Modelica library. Means a (tool dependent) readout of all needed text strings into a *LibraryName*.pot-File. It contains: 
+- a header which concludes all necessary information
+![Header of a *.pot file](POT_Header.png)
+
+- for all original strings entries with following structure: 
+```
+    msgctxt "Hydraulics.Basics.TankOverflow"
+    msgid "Volume Flow Port A"
+    msgstr ""
+```
+
+Where *msgctxt* contains the name of the Modelica Class and *msgid* contains the textstring which is situated in this name space and shall be translated. Please regard that if a *msgid* string (in this example "Volume Flow Port A") is given more than once in a namespace, all occurrences are translated with the same translation!
+
+It shall be mandatory to use for *msgctxt* the full name of a class:
+- so a specific translation depending on the content in this namespace is possible, as every textstring may differ depending on the context,
+- using the class makes it easier for the Modelica tool to copy and re-arrange models and packages without loosing the already existing language information.
+
+The following Modelica constructs shall be read out / translated: 
+* description strings
+* strings in following annotations:
+  * Text.string, Text.textString
+  * missingInnerMessage, obsolete, unassignedMessage 
+  * Dialog.[group|tab] 
+  * Dialog.[load|save]Selector.[caption|filter] 
+  * Documentation.[info|revisions]
+  * Figure.title, Plot.title, Curve.legend
+
+Having read out the file it just needs to be changed, if the library is changed (e.g. commented parts are added or changed or when having changed the Modelica name of the text strings containing element). 
+
+#### 2. Translation of the template file into a language file
+Starting from the template file create a \*.po file for each needed language (e.g. german: *LibraryName*.de.po-File).
+
+Edit the *msgstr* with the translation in the wanted language (here for german: `` Hydraulics.de.po``) :
+```
+    msgctxt "Hydraulics.Basics.TankOverflow"
+    msgid "Volume Flow Port A"
+    msgstr "Volumenstrom an Anschluss A"

```

Comment:
This was the only example I found and it is not a literal translation:
Why does it have capital letters in English?
Better: "Volume flow at port A"

Note, I also added "at" to the English version, in order to have a literal translation to "an" in German.

Reply:
I don't think we need to worry too much about the markdown files now that final touches are being made to the specification changes before entering the _Under Evaluation_ stage.

Reply:
Ok, fine. I just tried to understand how to use the proposal and this was the only example I found. 
I was not aware that this Markdown will not go into "frozen mode" (Under evaluation).

Reply:
My understanding is that it goes into "frozen mode", but that it will only serve as historical background for what ended up as specification changes.

The example you're looking for starts with: _An entry for translating the description of…_

Reply:
I wouldn't say that the markdown is completely frozen, but to me the main part is to provide the example in the actual specification text - and I agree that it should be expanded.
I thus added a complete class to show what is possible. (Unfortunately I couldn't get Swedish characters to work in that context.)

Reply:
So I would propose to resolve this and continue example-discussion in https://github.com/modelica/ModelicaSpecification/pull/2956#discussion_r983346323_